### PR TITLE
kusion watch: filter out unchanged resources

### PR DESCRIPTION
**Pull request by howieyuen**

### CODE REVIEW

**Review by markliby — APPROVED** [review]

File: pkg/kusionctl/cmd/apply/options.go [hunk 1]
```diff
@@ -314,38 +314,45 @@ func Apply(
...
-	project *projectstack.Project,
-	stack *projectstack.Stack,
+	changes *opsmodels.Changes,
 	out io.Writer,
 ) error {
 	if o.DryRun {
 		fmt.Fprintln(out, "NOTE: Watch doesn't work in DryRun mode")
 		return nil
 	}
-	if !o.Watch {
```
Comment (SparkYuan) [inline pkg/kusionctl/cmd/apply/options.go:332]: We don't need to check this watch option anymore?

Comment (howieyuen) [reply pkg/kusionctl/cmd/apply/options.go:332]: already checked in  line 152, so I remove the dup code

**Review by SparkYuan — DISMISSED** [review]
lgtm

**Review by SparkYuan — APPROVED** [review]
lgtm

**Review by markliby — APPROVED** [review]

**Pull request merged**